src/Symfony/Bundle/FrameworkBundle/Routing/RouteCollectionBuilder.php

```diff
@@ -0,0 +1,393 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Routing;
+
+use Symfony\Component\Config\Loader\Loader;
+use Symfony\Component\Config\Resource\ResourceInterface;
+use Symfony\Component\Routing\RouteCollection;
+
+/**
+ * Helps add and import routes into a RouteCollection..
+ *
+ * @author Ryan Weaver <ryan@knpuniversity.com>
+ */
+class RouteCollectionBuilder
+{
+    private $loader;
+
+    /**
+     * A mixture of different objects that hold routes.
+     *
+     * @var Route[]|RouteCollectionBuilder[]|RouteCollection[]
+     */
+    private $routes = array();
+
+    private $defaults = array();
+
+    private $prefix;
+
+    private $host;
+
+    private $condition;
+
+    private $requirements = array();
+
+    private $options = array();
+
+    private $schemes;
+
+    private $methods;
+
+    private $resources = array();
+
+    private $controllerClass;
+
+    /**
+     * @param Loader $loader
+     */
+    public function __construct(Loader $loader)
+    {
+        $this->loader = $loader;
+    }
+
+    /**
+     * Import an external routing resource, like a file.
+     *
+     * Returns a RouteCollectionBuilder so you can continue to tweak options on the routes.
+     *
+     * @param mixed $resource
+     * @param string $prefix
+     * @param string $type
+     * @return RouteCollectionBuilder
+     */
+    public function import($resource, $prefix = null, $type = null)
+    {
+        /** @var RouteCollection $subCollection */
+        $subCollection = $this->loader->import($resource, $type);
+
+        // turn this into a RouteCollectionBuilder
+        $builder = new RouteCollectionBuilder($this->loader);
+        $builder->setPrefix($prefix);
+        $builder->addRouteCollection($subCollection);
+        $this->routes[] = $builder;
+
+        return $builder;
+    }
+
+    /**
+     * Adds a route and returns it for future modification.
+     *
+     * @param string $path          The route path
+     * @param string $controller    The route controller string
+     * @param string $name          The name to give this route
+     * @return Route
+     */
+    public function add($path, $controller, $name = null)
```

Comment:
We could potentially also have a `get()` - I was trying to minimize the API, but it would allow you to do things like mutate imported routes.